### PR TITLE
[Feature] Support video input and output in restoration demo

### PR DESCRIPTION
**Pull request by ckkelvinchan**
## Motivation
When inferencing a video model, one would need to convert the video (e.g., `.mp4`) to images, and then load the images. This causes inconvenience if one wants to quickly try the result.

## Modification
This PR supports using video format as input and output. One could simply replace the directory with the path to the video.

**Note**: When saving the outputs as a video, it introduces video compression, which would lower the visual quality.

### CODE REVIEW

File: demo/restoration_video_demo.py [hunk 1]
```diff
@@ -28,26 +33,51 @@ def parse_args():
...
+        help='maximum sequence length if recurrent framework is used')
     parser.add_argument('--device', type=int, default=0, help='CUDA device id')
     args = parser.parse_args()
     return args
 
 
 def main():
+    """ Demo for video restoration models.
+
+    Note that we accept video as input(output), when 'input_dir'('output_dir')
```
Comment (Yshuo-Li) [inline demo/restoration_video_demo.py:49]: input (output)
'input_dir' ('output_dir')

File: demo/restoration_video_demo.py [hunk 2]
```diff
@@ -28,26 +33,51 @@ def parse_args():
...
+    is set to the path to the video. But using videos introduces video
+    compression, which lower the visual quality. If you want actual quality,
```
Comment (Yshuo-Li) [inline demo/restoration_video_demo.py:51]: which lowers ...

File: demo/restoration_video_demo.py [hunk 3]
```diff
@@ -28,26 +33,51 @@ def parse_args():
...
+    please save them as separate images (.png).
```
Comment (Yshuo-Li) [inline demo/restoration_video_demo.py:52]: please save them as separated images (.png).

**Review by Yshuo-Li — COMMENTED** [review]
Several lines are not covered by tests

**Comment by ckkelvinchan** [comment]
> Several lines are not covered by tests

It is tested in GPU only. Therefore it is not revealed in the CI report.

**Review by Yshuo-Li — APPROVED** [review]

**Review by Yshuo-Li — APPROVED** [review]

**Pull request merged**